src/Altinn.AccessManagement.Integration/Clients/Altinn2RightsClient.cs

```diff
@@ -0,0 +1,90 @@
+{
+    /// <summary>
+    /// Client for getting Altinn roles from AltinnII SBL Bridge
+    /// </summary>
+    public class Altinn2RightsClient : IAltinn2RightsClient
+    {
+        private readonly SblBridgeSettings _sblBridgeSettings;
+        private readonly HttpClient _client;
+        private readonly ILogger _logger;
+        private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AltinnRolesClient"/> class
+        /// </summary>
+        /// <param name="httpClient">HttpClient from default httpclientfactory</param>
+        /// <param name="sblBridgeSettings">the sbl bridge settings</param>
+        /// <param name="logger">the logger</param>
+        public Altinn2RightsClient(HttpClient httpClient, IOptions<SblBridgeSettings> sblBridgeSettings, ILogger<AltinnRolesClient> logger)
+        {
+            _sblBridgeSettings = sblBridgeSettings.Value;
+            _logger = logger;
+            _client = httpClient;
+            _serializerOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        /// <inheritdoc />
+        public async Task<DelegationCheckResponse> PostDelegationCheck(int userId, string partyId, string serviceCode, string serviceEditionCode)
+        {
+            try
+            {
+                UriBuilder uriBuilder = new UriBuilder($"{_sblBridgeSettings.BaseApiUrl}authorization/api/rights/delegation/userdelegationcheck?userId={userId}&partyId={partyId}&serviceCode={serviceCode}&serviceEditionCode={serviceEditionCode}");
+                _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                HttpResponseMessage response = await _client.GetAsync(uriBuilder.Uri);
+                string rights = await response.Content.ReadAsStringAsync();
+
+                DelegationCheckResponse delegationCheckResponse = new DelegationCheckResponse();
+
+                if (response.StatusCode == System.Net.HttpStatusCode.OK)
+                {
+                    delegationCheckResponse.RightDelegationCheckResults = JsonSerializer.Deserialize<List<RightDelegationCheckResult>>(rights, _serializerOptions);
+                    return delegationCheckResponse;
+                }
+
+                if (response.StatusCode == System.Net.HttpStatusCode.BadRequest)
+                {
+                    SBLUserDelegationCheckError validationError = JsonSerializer.Deserialize<SBLUserDelegationCheckError>(rights, _serializerOptions);
+                    _logger.LogError($"AccessManagement // Altinn2RightsClient // PostDelegationCheck // Unexpected HttpStatusCode: {response.StatusCode}");
+                    foreach (KeyValuePair<string, List<string>> modelState in validationError.ModelState)
+                    {
+                        string thing = string.Empty;
```

Comment:
StyleCop: bad naming 'thing'

Reply:
It's also unused as it's still modelState.Value.First() which is added. 
the foreach modelState.Value string concatenation can also be replaced by a simple
`string.Join(", ", modelState.Value);`